docs/sdk/js/cross-contract/callbacks.md

```diff
@@ -0,0 +1,86 @@
+---
+sidebar_position: 2
+title: Callbacks
+---
+
+# Callbacks
+
+NEAR Protocol is a sharded, proof-of-stake blockchain that behaves differently than proof-of-work blockchains. When interacting with a native Rust (compiled to Wasm) smart contract, cross-contract calls are asynchronous. Callbacks are used to either get the result of a cross-contract call or tell if a cross-contract call has succeeded or failed.
+
+## Calculator Example
+
+A callback method can be declared in your contract class as a regular method decorated with the `call({})` decorator. Be sure to pass in the `privateFunction: true` option to the decorator. This will ensure that the method is only callable by the contract itself.
+
+For example, let's assume the calculator is deployed on `calc.near`, we can use the following:
+
+```js
+@NearBindgen({})
+export class CalculatorCallerContract {
+  @call({})
+  sum_a_b({ a, b }) {
+    let calculatorAccountId = "calc.near";
+    // Call the method `sum` on the calculator contract.
+    // Any unused GAS will be attached since the default GAS weight is 1.
+    // Attached deposit is defaulted to 0.
+    return NearPromise
+            .new(calculatorAccountId)
+            .functionCall("sum", { a, b }, BigInt(0), BigInt(100000000000000));
+  }
+}
+```
+
+## Allowlist Example
+
+Next we'll look at a simple cross-contract call that is made to an allowlist smart contract, returning whether an account is in the list or not.
+
+The common pattern with cross-contract calls is to call a method on an external smart contract, use `.then` syntax to specify a callback, and then retrieve the result or status of the promise. The callback will typically live inside the same, calling smart contract. There's a special decorator parameter used for protecting the callback function, which is [`privateFunction: true`](https://docs.rs/near-sdk-core/latest/near_sdk_core/struct.AttrSigInfo.html#structfield.is_private). We'll see this pattern in the example below.
+
+The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach with `NearPromise`. 
+
+```js
+@NearBindgen({})
+export class ExtAllowlist {
+    // ...
+
+    @call({})
+    is_allowlisted({ staking_pool_account_id }) {
+        return this.allowlist.get(staking_pool_account_id) != null;
+    };
+}
+```
+
+After creating the class, we'll show a simple flow that will make a cross-contract call to the allowlist smart contract, asking if the account `idea404.testnet` is allowlisted. 
+
+```js
+@NearBindgen({})
+export class Contract {
+    @call({})
```

Comment:
same question about `privateFunction` ?

Reply:
In this case the example is below. I'll highlight in the text that setting the function to private is what you want to do for the callback method specifically.